OpenSourcePolitics/decidim-app
# Feat/add decypted private body to extra fields

🎩 Description
Add new column on ProposalExtraField to store decrypted private body date, and add rake task to update existing data.

📌 Related Issues
[Odoo card](https://opensourcepolitics.odoo.com/web#id=3035&cids=1&menu_id=325&action=471&model=project.task&view_type=form)

Testing
As an admin, go to Decidim Awesome
Go to private_fields of Proposals custom fields and add a custom field, for example a radio group applicable on processes proposals (see screenshot)
Go to a process, than to proposals and add a new proposal
In rails console, check that a proposal extra field has been created with `Decidim::DecidimAwesome::ProposalExtraField.last`, and that it contains a decrypted_private_body that looks like `"<xml><dl class=\"decidim_awesome-custom_fields\" data-generator=\"decidim_awesome\  (........) </dl></xml>"`

To test the rake task:

1. Go to rails console and set the decrypted_private_body of proposal extra field previously created to nil with `Decidim::DecidimAwesome::ProposalExtraField.last.update_columns(decrypted_private_body: nil)` 
2. Run the rake task with `bundle exec rake decidim:set_decrypted_private_body` 
3. Check in the console that the previous proposal extra field has a decrypted_private_body again

Tasks
[X ] Add specs


### :camera: Screenshots

<img width="1136" alt="one" src="https://github.com/user-attachments/assets/4c28d01c-4bb9-4e6e-bc11-300c7d23d70f">
